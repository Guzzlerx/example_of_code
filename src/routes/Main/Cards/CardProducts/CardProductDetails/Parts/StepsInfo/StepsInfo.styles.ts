import styled from 'styled-components';
import { COLOR_PALETTE } from 'constants/colors/colors';

export const StepsInfoContainer = styled.section`
  display: flex;
  flex-direction: column;
  align-items: baseline;
  width: 100%;
  margin: 72px 0 87px 0;
`;

export const StepsWrapper = styled.div`
  display: flex;
  gap: 38px;
`;

export const StepWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: baseline;
  text-align: left;
  width: 292px;
`;

export const StepTitle = styled.div`
  font-size: 20px;
  line-height: 28px;
  margin-top: 24px;
`;

export const StepSubtitle = styled.div`
  font-size: 14px;
  line-height: 20px;
  color: ${COLOR_PALETTE.gray700};
  margin-top: 8px;
`;
