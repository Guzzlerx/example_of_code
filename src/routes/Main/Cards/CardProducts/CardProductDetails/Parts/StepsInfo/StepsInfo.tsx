import { useTranslation } from 'react-i18next';
import { AIcon } from 'components/AIcon/AIcon';
import { ATitle } from 'components/ATitle/ATitle';
import {
  StepSubtitle,
  StepTitle,
  StepWrapper,
  StepsInfoContainer,
  StepsWrapper,
} from './StepsInfo.styles';

export const StepsInfo = () => {
  const { t } = useTranslation('cards');

  const translations = {
    title: t('card_details.steps_info.title'),
    firstStepOrder: t('card_details.steps_info.first_step.order'),
    firstStepTitle: t('card_details.steps_info.first_step.title'),
    firstStepSubtitle: t('card_details.steps_info.first_step.subtitle'),
    secondStepOrder: t('card_details.steps_info.second_step.order'),
    secondStepTitle: t('card_details.steps_info.second_step.title'),
    secondStepSubtitle: t('card_details.steps_info.second_step.subtitle'),
    thirdStepOrder: t('card_details.steps_info.third_step.order'),
    thirdStepTitle: t('card_details.steps_info.third_step.title'),
    thirdStepSubtitle: t('card_details.steps_info.third_step.subtitle'),
  };

  const stepsData = [
    {
      order: translations.firstStepOrder,
      title: translations.firstStepTitle,
      subtitle: translations.firstStepSubtitle,
    },
    {
      order: translations.secondStepOrder,
      title: translations.secondStepTitle,
      subtitle: translations.secondStepSubtitle,
    },
    {
      order: translations.thirdStepOrder,
      title: translations.thirdStepTitle,
      subtitle: translations.thirdStepSubtitle,
    },
  ];

  return (
    <StepsInfoContainer>
      <ATitle size={20} lineHeight={28} weight={700} marginBottom={40}>
        {translations.title}
      </ATitle>
      <StepsWrapper>
        {stepsData.map(({ order, title, subtitle }) => (
          <StepWrapper key={order}>
            <AIcon iconTitle={order} variant="stepsInfo">
              {order}
            </AIcon>
            <StepTitle>{title}</StepTitle>
            <StepSubtitle>{subtitle}</StepSubtitle>
          </StepWrapper>
        ))}
      </StepsWrapper>
    </StepsInfoContainer>
  );
};
