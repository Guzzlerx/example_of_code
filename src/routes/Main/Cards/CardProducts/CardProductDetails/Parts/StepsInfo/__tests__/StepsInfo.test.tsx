import { RenderResult, render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import i18n from 'utils/languages/i18nForTests';
import { StepsInfo } from '../StepsInfo';

describe('StepsInfo', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(
      <I18nextProvider i18n={i18n}>
        <StepsInfo />
      </I18nextProvider>,
    );
  });

  it('component should render', () => {
    expect(renderResult.container).toBeInTheDocument();
  });

  it('snapshot', () => {
    expect(renderResult.container).toMatchSnapshot();
  });
});
