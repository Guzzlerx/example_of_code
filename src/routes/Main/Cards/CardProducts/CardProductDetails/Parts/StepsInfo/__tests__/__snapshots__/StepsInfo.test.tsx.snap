// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepsInfo snapshot 1`] = `
<div>
  <section
    class="sc-dkrFOg eZyYpP"
  >
    <h2
      class="sc-gswNZR eugAJR"
      size="20"
    >
      Get your card in three steps
    </h2>
    <div
      class="sc-hLBbgP iAYQMH"
    >
      <div
        class="sc-eDvSVe exIElQ"
      >
        <div
          class="sc-bcXHqe bioYJS"
          title="1"
        >
          1
        </div>
        <div
          class="sc-jSUZER cgFgaH"
        >
          Press “Get a card” button
        </div>
        <div
          class="sc-gKPRtg eZQHHz"
        >
          No need to enter additional data
        </div>
      </div>
      <div
        class="sc-eDvSVe exIElQ"
      >
        <div
          class="sc-bcXHqe bioYJS"
          title="2"
        >
          2
        </div>
        <div
          class="sc-jSUZER cgFgaH"
        >
          Wait for approval
        </div>
        <div
          class="sc-gKPRtg eZQHHz"
        >
          We’ll send an email with decision in 2 min
        </div>
      </div>
      <div
        class="sc-eDvSVe exIElQ"
      >
        <div
          class="sc-bcXHqe bioYJS"
          title="3"
        >
          3
        </div>
        <div
          class="sc-jSUZER cgFgaH"
        >
          Get your card
        </div>
        <div
          class="sc-gKPRtg eZQHHz"
        >
          Pick up your card in the branch in 2 days
        </div>
      </div>
    </div>
  </section>
</div>
`;
